apps/aot/src/app/events/[year]/leaderboard/_components/overall-leaderboard.tsx

```diff
@@ -1,14 +1,47 @@
 import { Prisma, prisma } from '@repo/db';
 import { LEADERBOARD_RANKING_LIMIT } from '../constants';
-import { DataTableLeaderboard } from '@repo/ui/components/data-table-leaderboard';
-import { overallLeaderboardColumns, type OverallLeaderboardEntry } from './columns';
 import { getNextAdventDay } from '~/utils/time-utils';
 import { redisClient } from '@repo/redis';
 import { getAotChallengeIdsSoFar } from '../getAotChallengeIds';
 
-export const dynamic = 'force-dynamic';
+interface OverallLeaderboardEntry {
+  score: number | bigint;
+  name: string;
+  image: string | null;
+}
+
+export async function getOverallTableData(leaderboard: OverallLeaderboardEntry[]) {
+  const userData = await getUserData(leaderboard);
+  const rows = leaderboard.map((r) => {
+    const user = userData.find((u) => u.name === r.name);
+    if (user === undefined) {
+      throw new Error(`Leaderboard entry for ${r.name} does not have a matching user record`);
+    }
+    return {
+      ...user,
+      ...r,
+      score: Number(r.score),
+    };
+  });
+  return rows;
+}
```

Comment:
In tandem with the suggestion below, I think this can be removed, and then you can use the leaderboard data to build the table, and for each row just directly reference the map we built below. Every user will have a guaranteed entry in the map.